modules/drivers/druid-jdbc/src/metabase/driver/druid_jdbc.clj

```diff
@@ -0,0 +1,183 @@
+(defmethod sql.qp/date [:druid-jdbc :week] [_ _ expr] (date-trunc :week expr))
+(defmethod sql.qp/date [:druid-jdbc :month] [_ _ expr] (date-trunc :month expr))
+(defmethod sql.qp/date [:druid-jdbc :quarter] [_ _ expr] (date-trunc :quarter expr))
+(defmethod sql.qp/date [:druid-jdbc :year] [_ _ expr] (date-trunc :year expr))
+
+(defn- time-extract [unit expr] [:time_extract expr (h2x/literal unit)])
+
+(defmethod sql.qp/date [:druid-jdbc :minute-of-hour] [_ _ expr] (time-extract :minute expr))
+(defmethod sql.qp/date [:druid-jdbc :hour-of-day] [_ _ expr] (time-extract :hour expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-week] [_ _ expr] (time-extract :dow expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-month] [_ _ expr] (time-extract :day expr))
+(defmethod sql.qp/date [:druid-jdbc :day-of-year] [_ _ expr] (time-extract :doy expr))
+(defmethod sql.qp/date [:druid-jdbc :week-of-year] [_ _ expr] (time-extract :week expr))
+(defmethod sql.qp/date [:druid-jdbc :month-of-year] [_ _ expr] (time-extract :month expr))
+(defmethod sql.qp/date [:druid-jdbc :quarter-of-year] [_ _ expr] (time-extract :quarter expr))
+
+(defmethod sql.qp/current-datetime-honeysql-form :druid-jdbc
+  [_]
+  [:raw "CURRENT_TIMESTAMP"])
+
+(defmethod sql.qp/add-interval-honeysql-form :druid-jdbc
+  [_ hsql-form amount unit]
+  [:TIMESTAMPADD (h2x/identifier :type-name unit) (h2x/->integer amount) hsql-form])
+
+;; Query used for field values computation is not compatible with COMPLEX<...> types.
+(defmethod driver/field-values-compatible? :druid-jdbc
+  [_driver {:keys [database_type] :as _field}]
+  (not (re-find #"COMPLEX<" (str database_type))))
+
+(defmethod sql-jdbc.conn/data-source-name :druid-jdbc
+  [_driver {:keys [host port] :as _details}]
+  (format "druid-%s-%s"
+          ;; remove protocol
+          (or (re-find #"(?<=\w://).*" host)
+              host)
+          port))
+
+(defmethod sql-jdbc.execute/set-parameter [:druid-jdbc ZonedDateTime]
+  [driver ps i t]
+  (sql-jdbc.execute/set-parameter driver ps i (t/format "yyyy-MM-dd HH:mm:ss" t)))
+
+(defmethod unprepare/unprepare-value [:druid-jdbc ZonedDateTime]
+  [_ t]
+  (format "'%s'" (t/format "yyyy-MM-dd HH:mm:ss" t)))
+
+(defmethod sql.qp/json-query :druid-jdbc
+  [_driver unwrapped-identifier nfc-field]
+  (assert (h2x/identifier? unwrapped-identifier)
+          (format "Invalid identifier: %s" (pr-str unwrapped-identifier)))
+  (let [nfc-path          (:nfc-path nfc-field)
+        parent-identifier (sql.qp.u/nfc-field->parent-identifier unwrapped-identifier nfc-field)
+        ;; Druid json functions reference: https://druid.apache.org/docs/latest/querying/math-expr#json-functions
+        operator (if (isa? (:base-type nfc-field) :type/Array)
+                   ::json_query
+                   ::json_value)]
+    ;; TODO: Is there a way to avoid using :raw? If I do literal would that be correct?
+    [operator parent-identifier [:raw "'" (str/join "." (into ["$"] (rest nfc-path))) "'"]]))
```

Comment:
yes, `h2x/literal` will add the single quotes around it for you, also it will (attempt to) escape the string you're quoting which will make this a lot safer against SQL injection if we're accidentally using it with a user-supplied string

Reply:
```suggestion
    ;; TODO: Is there a way to avoid using :raw? If I do literal would that be correct?
    [operator parent-identifier (h2x/literal (str/join "." (into ["$"] (rest nfc-path))))]))
```

Reply:
To be clear the performance difference is not really important (just a nice bonus), cons` being clearer code is the main reason I suggested it

Reply:
Thanks for the suggestion and also for showing me `criterium`.